wordpress-mobile/WordPress-Android
# App Data Coverage: Add site picker track events

Parent #16343 

This PR adds track events for SitePicker actions.

- When the My Site site picker dropdown is tapped
- When the site picker is displayed
- When the site picker is dismissed
- When the edit button is tapped
- When the edit is done
- When the add site button is tapped
- When a search is performed (each time a letter is entered too)
- A visibility is changed for a site

**Note**: To stay consistent with iOS, these events are tracked with prefix “site_switcher”

To test:

1. Launch the app and sign in with an account with multiple blogs enabled]
2. Navigate to My Site tab
3. Tap the down arrow on the site header info section to open the Site Picker
4. ✅ Verify the logs contain: `🔵 Tracked: my_site_site_switcher_tapped`
5. ✅ Verify the logs contain: `🔵 Tracked: site_switcher_displayed`
6. Perform a search
7. ✅ Verify the logs contain: `🔵 Tracked: site_switcher_search_performed`
8. Tap the More menu on the menu bar and tap Show/Hide sites
9. ✅ Verify the logs contain:  `🔵 Tracked: site_switcher_toggled_edit_tapped, Properties: {"state":"edit"}`
10. Tap a blog row to highlight it
11. Tap the “HIDE” action button to set the visibility to hidden
12. ✅ Verify the logs contain: `🔵 Tracked: site_switcher_toggle_blog_visible, Properties: {"blog_id":"site-id","visible":"0"}`
13. Tap the More menu on the menu bar and tap Show/Hide sites
14. Long tap a site to enable action mode
15. Tap the row of the site you just hide to select it
16. Tap the “SHOW” action button to set the visibility to visible
17. ✅ Verify the logs contain: `🔵 Tracked: site_switcher_toggle_blog_visible, Properties: {"blog_id":"site-id","visible":"1"}`
18. ✅ Verify the logs contain: `🔵 Tracked: site_switcher_toggled_edit_tapped, Properties: {"state":"done"
19. Tap the + button to add a site
20. ✅ Verify the logs contain: `🔵 Tracked: site_switcher_add_site_tapped`
21. Tap the 'X' button to leave the Site Picker
22. ✅ Verify the logs contain: `🔵 Tracked: site_switcher_dismissed`

## Regression Notes
1. Potential unintended areas of impact
Events are not tracked properly

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual Testing

3. What automated tests I added (or what prevented me from doing so) N/A

PR submission checklist:
